File: demo/demo_cc/tools/demo02_record_read.cc
#include "cyber/cyber.h"
#include "cyber/record/record_message.h"
#include "cyber/record/record_reader.h"

using apollo::cyber::record::RecordMessage;
using apollo::cyber::record::RecordReader;

int main(int argc, char *argv[]) {
  apollo::cyber::Init(argv[0]);

  AINFO << "record 读操作演示...";
  RecordReader reader("/apollo/cyber/demo_cc/tools/mytest.record");
  // 读取话题相关数据
  std::string channel = "chatter";
  uint64_t msg_count = reader.GetMessageNumber(channel);
  AINFO << "总共" << msg_count << "条消息";
  std::string msg_type = reader.GetMessageType(channel);
  std::string msg_desc = reader.GetProtoDesc(channel);
  AINFO << "消息类型:" << msg_type;
  AINFO << "消息描述:" << msg_desc;
  // 读取消息内容
  RecordMessage msg;
  for (size_t i = 0; i < msg_count; i++) {
    if (reader.ReadMessage(&msg)) {
      AINFO << "第" << i << "条消息 -> "
            << "channel name: " << msg.channel_name
            << "; content: " << msg.content << "; msg time: " << msg.time;
    } else {
      AINFO << "消息读取失败.....";
    }
  }

  return 0;
}

File: demo/demo_py/tools/demo02_record_read.py
#!/usr/bin/env python3
"""  
    需求：将 student 消息写入到磁盘文件
    实现：
        1.创建读取对象；
        2.读消息；
        3.关闭文件流。


"""
from cyber.python.cyber_py3 import cyber
from cyber.python.cyber_py3 import record
from cyber.demo_base_proto.student_pb2 import Student

if __name__ == "__main__":
    cyber.init()
    print("record read ....")
    reader = record.RecordReader("/apollo/cyber/demo_py/tools/mytest_py.record")
    msgs = reader.read_messages()
    count = 0
    for msg in msgs:
        count += 1
        print("=" * 20)
        print("data NO.%d" % count)
        print("channel_name:%s" % msg[0])
        print("msg num %d" % reader.get_messagenumber(msg[0]))
        print("data:%s" % msg[1])
        print("data type:%s" % msg[2])
        print("time %d" % msg[3])

        msg_new = Student()
        msg_new.ParseFromString(msg[1])
        print(msg_new)